ir/dbprint.cpp

```diff
@@ -68,12 +68,36 @@ void IR::InstantiatedBlock::dbprint(std::ostream &out) const {
 
 void IR::Annotation::dbprint(std::ostream &out) const {
     out << '@' << name;
-    const char *sep = "(";
-    for (auto e : expr) {
-        out << sep << e;
-        sep = ", ";
-    }
-    if (*sep != '(') out << ')';
+    if (needsParsing()) out << "<unparsed>";
+    const char *open = structured ? "[" : "(";
+    const char *close = structured ? "]" : ")";
+
+    const char *sep = open;
+    std::visit(
+        [&](const auto &body) {
+            using T = std::decay_t<decltype(body)>;
+            if constexpr (std::is_same_v<T, IR::Vector<IR::AnnotationToken>>) {
+                for (auto e : body) {
+                    out << sep << e;
+                    sep = ", ";
```

Comment:
for unparsed annotations, the separator should be space or nothing.  Ideally only have a space where needed to not merge adjacent tokens, though unconditional space might not be too bad.